sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -118,6 +118,9 @@ class SessionCatalog(
   }
 
   def dropDatabase(db: String, ignoreIfNotExists: Boolean, cascade: Boolean): Unit = {
+    if (db == "default") {
```

Comment:
So far, the best way I found is to override `caseSensitiveAnalysis` to `false` in `conf` of `HiveSessionState`. 
```scala
  override lazy val conf: SQLConf = new SQLConf {
    override def caseSensitiveAnalysis: Boolean = false
  }
```